inc/saihostintf.h

```diff
@@ -214,101 +219,124 @@ typedef enum _sai_hostif_trap_type_t
+    SAI_HOSTIF_TRAP_TYPE_IPV6_MLD_V1_V2 = 0x0000200a,
 
     /** default packet action is forward */
-    SAI_HOSTIF_TRAP_TYPE_IPV6_MLD_V1_REPORT = 0x0000200d,
+    SAI_HOSTIF_TRAP_TYPE_IPV6_MLD_V1_REPORT = 0x0000200b,
 
     /** default packet action is forward */
-    SAI_HOSTIF_TRAP_TYPE_IPV6_MLD_V1_DONE = 0x0000200e,
+    SAI_HOSTIF_TRAP_TYPE_IPV6_MLD_V1_DONE = 0x0000200c,
 
     /** default packet action is forward */
-    SAI_HOSTIF_TRAP_TYPE_MLD_V2_REPORT = 0x0000200f,
+    SAI_HOSTIF_TRAP_TYPE_MLD_V2_REPORT = 0x0000200d,
+
+    /** Router traps custom range start */
+    SAI_HOSTIF_TRAP_TYPE_ROUTER_CUSTOM_RANGE_BASE = 0x0003000,
+
+    /* local IP traps */
 
     /**
      * @brief IP packets to local router IP address (routes with
      * #SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID = #SAI_SWITCH_ATTR_CPU_PORT)
-     * (default packet action is trap)
+     * (default packet action is drop)
      */
-    SAI_HOSTIF_TRAP_TYPE_IP2ME = 0x00002010,
+    SAI_HOSTIF_TRAP_TYPE_IP2ME = 0x00004000,
 
     /**
      * @brief SSH traffic (tcp dst port == 22) to local router IP address
-     * (default packet action is trap)
+     * (default packet action is drop)
      */
-    SAI_HOSTIF_TRAP_TYPE_SSH = 0x00002011,
+    SAI_HOSTIF_TRAP_TYPE_SSH = 0x00004001,
 
     /**
      * @brief SNMP traffic (udp dst port == 161) to local router IP address
-     * (default packet action is trap)
+     * (default packet action is drop)
      */
-    SAI_HOSTIF_TRAP_TYPE_SNMP = 0x00002012,
+    SAI_HOSTIF_TRAP_TYPE_SNMP = 0x00004002,
 
-    /** Router traps custom range start */
-    SAI_HOSTIF_TRAP_TYPE_ROUTER_CUSTOM_RANGE_BASE = 0x0003000,
+    /**
+    * @brief BGP traffic (tcp src port == 179 or tcp dst port == 179) to local
+    * router IP address (default packet action is drop)
+    */
+    SAI_HOSTIF_TRAP_TYPE_BGP = 0x00004003,
+
+    /**
+    * @brief BGPv6 traffic (tcp src port == 179 or tcp dst port == 179) to
+    * local router IP address (default packet action is drop)
+    */
+    SAI_HOSTIF_TRAP_TYPE_BGPV6 = 0x00004004,
+
+    /** Local IP traps custom range start */
+    SAI_HOSTIF_TRAP_TYPE_LOCAL_IP_CUSTOM_RANGE_BASE = 0x0005000,
 
     /* Pipeline exceptions */
 
     /**
      * @brief Packets size exceeds the router interface MTU size
-     * (default packet action is trap)
+     * (default packet action is drop)
      */
-    SAI_HOSTIF_TRAP_TYPE_L3_MTU_ERROR = 0x00004000,
+    SAI_HOSTIF_TRAP_TYPE_L3_MTU_ERROR = 0x00006000,
 
     /**
      * @brief Packets with TTL 0 or 1
-     * (default packet action is trap)
+     * (default packet action is drop)
      */
-    SAI_HOSTIF_TRAP_TYPE_TTL_ERROR = 0x00004001,
+    SAI_HOSTIF_TRAP_TYPE_TTL_ERROR = 0x00006001,
 
     /** Exception traps custom range start */
-    SAI_HOSTIF_TRAP_TYPE_CUSTOM_EXCEPTION_RANGE_BASE = 0x00005000,
+    SAI_HOSTIF_TRAP_TYPE_CUSTOM_EXCEPTION_RANGE_BASE = 0x00007000,
+
+    /**
+     * @brief End of trap types
+     */
+    SAI_HOSTIF_TRAP_TYPE_END,
```

Comment:
Shouldn't it be (SAI_HOSTIF_TRAP_USER_DEFINED_TYPE_END = SAI_HOSTIF_TRAP_TYPE_CUSTOM_EXCEPTION_RANGE_BASE + 0x1000) ?